modules/ansible/main.tf

```diff
@@ -66,6 +66,7 @@ resource "terraform_data" "trigger_ansible_vars" {
 
 resource "terraform_data" "execute_playbooks" {
   depends_on = [terraform_data.setup_ansible_host]
+  count      = var.ansible_vault_password != null ? 0 : 1
```

Comment:
at the end of the file include a moved block
moved {
from = terraform_data.execute_playbooks
to = terraform_data.execute_playbooks[0]
}